plasmapy/particles/ionization_state_collection.py

```diff
@@ -162,8 +166,11 @@ def __init__(
                     )
                 set_abundances = False
 
+        def return_none():
+            return None
+
         try:
-            self._pars = collections.defaultdict(lambda: None)
+            self._pars = collections.defaultdict(return_none)
```

Comment:
One of the things I read suggested that `defaultdict` objects cannot be pickled, so we might need to venture away from them.  Out of scope for this PR, though.